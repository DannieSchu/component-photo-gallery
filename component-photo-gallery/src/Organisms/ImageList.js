import React, { Component } from 'react';
import ImageItem from '../Molecules/ImageItem'
import images from '../data';

export default class ImageList extends Component {
    render() {
    return (    
        <ul>
            {images.map((creature, index) => 
            <ImageItem image = {creature} key={`${index} - ${creature.description}`} />)}
        </ul>    
    )
}
}