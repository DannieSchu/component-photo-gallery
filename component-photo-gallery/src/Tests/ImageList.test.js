import React from 'react'
import renderer from 'react-test-renderer';
import ImageList from '../Organisms/ImageList';



test('renders ImageList.js correctly', () => {
    const tree = renderer
      .create(<ImageList />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });