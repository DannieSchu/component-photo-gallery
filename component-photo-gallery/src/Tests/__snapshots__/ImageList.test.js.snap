// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ImageList.js correctly 1`] = `
<ul>
  <li
    className="flex-column"
  >
    <h3>
      UniWhal
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="narwhal"
        src="http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg"
      />
    </div>
    <p>
      A unicorn and a narwhal nuzzling their horns
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Rhino Family
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="rhino"
        src="https://images.unsplash.com/photo-1512636618879-bbe79107e9e3?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=bd9460ee6d1ddbb6b1ca7be86dfc4590&auto=format&fit=crop&w=1825&q=80"
      />
    </div>
    <p>
      Mother (or father) rhino with two babies
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Unicorn Head
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="unicorn"
        src="https://www.dhresource.com/0x0s/f2-albu-g5-M00-1A-11-rBVaI1hsIIiALxKzAAIHjSU3VkE490.jpg/wholesale-halloween-costume-prop-unicorn.jpg"
      />
    </div>
    <p>
      Someone wearing a creepy unicorn head mask
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      UniLego
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="unilego"
        src="https://images.unsplash.com/photo-1518946222227-364f22132616?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=4836a6fca62e7dce9324346bacfde085&auto=format&fit=crop&w=2534&q=80"
      />
    </div>
    <p>
      Lego figurine dressed in a unicorn outfit
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Basically a unicorn
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="narwhal"
        src="https://i.pinimg.com/736x/b4/61/06/b46106830b841017ea59870b27ec18dc--narwhals-a-unicorn.jpg"
      />
    </div>
    <p>
      A narwhal is basically a unicorn after all, right?
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      #truth
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="narwhal"
        src="https://i.pinimg.com/originals/16/cf/2a/16cf2a0b3fd51b9bee08bb6296193b75.jpg"
      />
    </div>
    <p>
      The truth behind narwhals
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Baby Rhino
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="rhino"
        src="https://secure.img1-ag.wfcdn.com/im/17007094/resize-h800%5Ecompr-r85/3589/35892451/Baby+Rhino+Figurine.jpg"
      />
    </div>
    <p>
      This is actually a figurine but it looks kinda real
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Cera
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="triceratops"
        src="https://vignette.wikia.nocookie.net/landbeforetime/images/c/c3/Cera_infobox.png/revision/latest?cb=20180422005003"
      />
    </div>
    <p>
      Three horns but still, horns. And who doesn't like The Land Before Time?
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Narwhal costume
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="narwhal"
        src="https://ae01.alicdn.com/kf/HTB18GwSQVXXXXaZaXXXq6xXFXXXh/Animal-Cosplay-Costume-Narwhal-Onesie-Mens-Womens-Cartoon-Whale-Pajamas.jpg"
      />
    </div>
    <p>
      A woman wearing a blue narwhal costume
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Rhino costume
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="rhino"
        src="https://www.shopmascot.com/image/cache/mascotnew/new196-800x800.jpg"
      />
    </div>
    <p>
      Mascots have to get their costumes somewhere
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Believe
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="unicorn"
        src="https://www.tinselbox.com/wp-content/uploads/2018/03/I-BELIEVE-IN-UNICORNS-FREE-PRINTABLE-WATERCOLOR-7-of-1.jpg"
      />
    </div>
    <p>
      I believe in unicorns, do you?
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Markhor
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="markhor"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/3/38/Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg/220px-Markhor_Schraubenziege_Capra_falconeri_Zoo_Augsburg-02.jpg"
      />
    </div>
    <p>
      These wild goats eat snakes, then secrete a foam that locals fight over for the antivenom properties -- how cool is that?
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Baby markhor
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="markhor"
        src="http://www.zooborns.com/.a/6a010535647bf3970b0223c84d5959200c-800wi"
      />
    </div>
    <p>
      Even the babies are adorable
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Mouflon
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="mouflon"
        src="http://cdn.modernfarmer.com/wp-content/uploads/2014/08/28476658_9c97f35096_o.jpg"
      />
    </div>
    <p>
      Those horns though
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Addax
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="addax"
        src="http://cdn.modernfarmer.com/wp-content/uploads/2014/08/addax.jpg"
      />
    </div>
    <p>
      This guy is basically extinct but survives well in captivity, so they're frequently found in zoos
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Baby mouflon
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="mouflon"
        src="https://cbsnews3.cbsistatic.com/hub/i/r/2013/03/05/5b414225-a645-11e2-a3f0-029118418759/thumbnail/620x350/2d4cf24685b45c22912e64d2004fec8d/Baby_Mouflon_Wild_Sheep.jpg"
      />
    </div>
    <p>
      The cuteness that is a baby mouflon asleep
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Happy Jackson's Chameleon
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="chameleon"
        src="http://cdn.modernfarmer.com/wp-content/uploads/2014/08/chameleon.jpg"
      />
    </div>
    <p>
      These are really common in Hawaii
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Serious Jackson's Chameleon
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="chameleon"
        src="https://imgc.allpostersimages.com/img/print/posters/dlillc-jackson-s-chameleon_a-G-13448768-14258384.jpg"
      />
    </div>
    <p>
      This one is very serious.
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Horned Lizard
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="lizard"
        src="https://www.nps.gov/band/learn/nature/images/short-horned-lizard-open-mouth-18.jpg?maxwidth=650&autorotate=false"
      />
    </div>
    <p>
      Fave food: ants
    </p>
  </li>
  <li
    className="flex-column"
  >
    <h3>
      Smaug
    </h3>
    <div
      className="image-container"
    >
      <img
        alt="dragon"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/df/Smaug_par_David_Demaret.jpg/290px-Smaug_par_David_Demaret.jpg"
      />
    </div>
    <p>
      Fan illustration of Smaug from 'The Hobbit'
    </p>
  </li>
</ul>
`;
